source/isaaclab_tasks/isaaclab_tasks/manager_based/manipulation/dexsuite/mdp/curriculums.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2022-2025, The Isaac Lab Project Developers (https://github.com/isaac-sim/IsaacLab/blob/main/CONTRIBUTORS.md).
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+from __future__ import annotations
+
+import torch
+from collections.abc import Sequence
+from typing import TYPE_CHECKING
+
+from isaaclab.assets import Articulation, RigidObject
+from isaaclab.envs import mdp
+from isaaclab.managers import ManagerTermBase, SceneEntityCfg
+from isaaclab.utils.math import combine_frame_transforms, compute_pose_error
+
+if TYPE_CHECKING:
+    from isaaclab.envs import ManagerBasedRLEnv
+
+
+def initial_final_interpolate_fn(env: ManagerBasedRLEnv, env_id, data, iv, fv, difficulty_term_str):
+    """
+    Interpolate between initial value iv and final value fv, for any arbitrarily
+    nested structure of lists/tuples in 'data'. Scalars (int/float) are handled
+    at the leaves.
+    """
+    # get the fraction scalar on the device
+    difficulty_term: DifficultyScheduler = getattr(env.curriculum_manager.cfg, difficulty_term_str).func
+    frac = difficulty_term.difficulty_frac
+    if frac < 0.1:
+        return mdp.modify_env_param.NO_CHANGE
+
+    # convert iv/fv to tensors, but we'll peel them apart in recursion
+    iv_t = torch.tensor(iv, device=env.device)
+    fv_t = torch.tensor(fv, device=env.device)
+
+    return recurse(iv_t.tolist(), fv_t.tolist(), data, frac)
+
+
+def recurse(iv_elem, fv_elem, data_elem, frac):
```

Comment:
If this is a private helper, then should start with an "_". Would be good to add type-hints and docstring to its functionality.